src/main/java/bio/terra/service/snapshot/flight/create/CreateSnapshotPrimaryDataQueryAzureStep.java

```diff
@@ -0,0 +1,87 @@
+package bio.terra.service.snapshot.flight.create;
+
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import bio.terra.grammar.Query;
+import bio.terra.grammar.azure.SynapseVisitor;
+import bio.terra.model.DatasetModel;
+import bio.terra.model.SnapshotRequestModel;
+import bio.terra.service.dataset.AssetSpecification;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.DatasetService;
+import bio.terra.service.dataset.flight.ingest.IngestUtils;
+import bio.terra.service.filedata.azure.AzureSynapsePdao;
+import bio.terra.service.snapshot.Snapshot;
+import bio.terra.service.snapshot.SnapshotDao;
+import bio.terra.service.snapshot.SnapshotService;
+import bio.terra.service.snapshot.SnapshotTable;
+import bio.terra.stairway.FlightContext;
+import java.sql.SQLException;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+public class CreateSnapshotPrimaryDataQueryAzureStep extends CreateSnapshotParquetFilesAzureStep
+    implements CreateSnapshotPrimaryDataQueryStep {
+
+  private final AzureSynapsePdao azureSynapsePdao;
+  private final SnapshotService snapshotService;
+  private final SnapshotDao snapshotDao;
+  private final SnapshotRequestModel snapshotReq;
+  private final DatasetService datasetService;
+  private final AuthenticatedUserRequest userRequest;
+  private String sourceDatasetDataSourceName;
+  private String targetDataSourceName;
+
+  public CreateSnapshotPrimaryDataQueryAzureStep(
+      AzureSynapsePdao azureSynapsePdao,
+      SnapshotDao snapshotDao,
+      SnapshotService snapshotService,
+      SnapshotRequestModel snapshotReq,
+      DatasetService datasetService,
+      AuthenticatedUserRequest userRequest) {
+    super(azureSynapsePdao, snapshotService, snapshotReq);
+    this.azureSynapsePdao = azureSynapsePdao;
+    this.snapshotService = snapshotService;
+    this.snapshotReq = snapshotReq;
+    this.snapshotDao = snapshotDao;
+    this.datasetService = datasetService;
+    this.userRequest = userRequest;
+  }
+
+  @Override
+  public Map<String, Long> createSnapshotParquetFiles(
+      List<SnapshotTable> tables, UUID snapshotId, FlightContext context)
+      throws SQLException, InterruptedException {
+    sourceDatasetDataSourceName = IngestUtils.getSourceDatasetDataSourceName(context.getFlightId());
+    targetDataSourceName = IngestUtils.getTargetDataSourceName(context.getFlightId());
+    Snapshot snapshot = snapshotDao.retrieveSnapshotByName(snapshotReq.getName());
+    return prepareQueryAndCreateSnapshot(context, snapshot, snapshotReq, datasetService);
+  }
```

Comment:
Sorry for the churn - I'm realizing I need to re-work this method that was originally intended to inherit from CreateSnapshotParquetFilesAzureStep.